ec2_resources.tf

```
@@ -106,13 +106,17 @@ resource "aws_security_group" "instance" {
   vpc_id = var.vpc_id
 
   # Allow SSH inbound
-  ingress {
-    description     = "SSH"
-    from_port       = 22
-    to_port         = 22
-    protocol        = "tcp"
-    cidr_blocks     = var.ssh_inbound_cidr
-    security_groups = var.ssh_inbound_security_group
+  dynamic "ingress" {
+    for_each = var.ssh_inbound_cidr
+
+    content {
+      description     = "SSH"
+      from_port       = 22
+      to_port         = 22
+      protocol        = "tcp"
+      cidr_blocks     = var.ssh_inbound_cidr
+      security_groups = var.ssh_inbound_security_group
+    }
```

I believe this would break if we have multiple entries in the `ssh_inbound_cidr` variable. What about using `count` instead? I mean:
```suggestion
  dynamic "ingress" {
    count = check_list_empty ? 0 : 1
    description     = "SSH"
    from_port       = 22
    to_port         = 22
    protocol        = "tcp"
    cidr_blocks     = var.ssh_inbound_cidr
    security_groups = var.ssh_inbound_security_group
```

---

I see what you mean! Yes, I think this is the way to go, thanks for catching that :)